--- target/linux/airoha/an7581/base-files/etc/board.d/02_network
+++ target/linux/airoha/an7581/base-files/etc/board.d/02_network
@@ -16,6 +16,9 @@ an7581_setup_interfaces()
 	gemtek,w1700k-ubi)
 		ucidef_set_interfaces_lan_wan "lan2 lan3 lan4" "wan"
 		;;
+	econet,xr1710g-ubi)
+		ucidef_set_interfaces_lan_wan "lan1 lan2 lan3" "wan"
+		;;
 	nokia,valyrian)
 		ucidef_set_interfaces_lan_wan "lan1 lan2 lan3 10g" "wan"
 		;;
